cpp/tests/strings/fixed_point_tests.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/fixed_point/fixed_point.hpp>
+#include <cudf/strings/convert/convert_fixed_point.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <tests/strings/utilities.h>
+
+#include <vector>
+
+struct StringsConvertTest : public cudf::test::BaseFixture {
+};
+
+template <typename T>
+class StringsFixedPointConvertTest : public StringsConvertTest {
+};
+
+TYPED_TEST_CASE(StringsFixedPointConvertTest, cudf::test::FixedPointTypes);
+
+TYPED_TEST(StringsFixedPointConvertTest, ToFixedPoint)
+{
+  using DecimalType = TypeParam;
+  using RepType     = cudf::device_storage_type_t<DecimalType>;
+  using fp_wrapper  = cudf::test::fixed_point_column_wrapper<RepType>;
+
+  cudf::test::strings_column_wrapper strings(
+    {"1234", "-876", "543.2", "-0.12", ".25", "-.002", "", "-0.0"});
+  auto results = cudf::strings::to_fixed_point(
+    cudf::strings_column_view(strings),
+    cudf::data_type{cudf::type_to_id<DecimalType>(), numeric::scale_type{-3}});
+  auto const expected =
+    fp_wrapper{{1234000, -876000, 543200, -120, 250, -2, 0, 0}, numeric::scale_type{-3}};
```

Comment:
I would add a test case that performs truncation.
```suggestion
  cudf::test::strings_column_wrapper strings(
    {"1234", "-876", "543.2", "-0.12", ".25", "-.002", "-.0027", "", "-0.0"});
  auto results = cudf::strings::to_fixed_point(
    cudf::strings_column_view(strings),
    cudf::data_type{cudf::type_to_id<DecimalType>(), numeric::scale_type{-3}});
  auto const expected =
    fp_wrapper{{1234000, -876000, 543200, -120, 250, -2, -2, 0, 0}, numeric::scale_type{-3}};
```